deepchem/models/tensorflow_models/lr.py

```diff
@@ -0,0 +1,231 @@
+    with graph.as_default():
+      epsilon = 1e-3  # small float to avoid dividing by zero
+      weighted_costs = []  # weighted costs for each example
+      gradient_costs = []  # costs used for gradient calculation
+
+      with TensorflowGraph.shared_name_scope('costs', graph, name_scopes):
+        for task in range(self.n_tasks):
+          task_str = str(task).zfill(len(str(self.n_tasks)))
+          with TensorflowGraph.shared_name_scope(
+              'cost_{}'.format(task_str), graph, name_scopes):
+            with tf.name_scope('weighted'):
+              weighted_cost = self.cost(output[task], labels[task],
+                                        weights[task])
+              weighted_costs.append(weighted_cost)
+
+            with tf.name_scope('gradient'):
+              # Note that we divide by the batch size and not the number of
+              # non-zero weight examples in the batch.  Also, instead of using
+              # tf.reduce_mean (which can put ops on the CPU) we explicitly
+              # calculate with div/sum so it stays on the GPU.
+              gradient_cost = tf.div(tf.reduce_sum(weighted_cost),
+                                     self.batch_size)
+              gradient_costs.append(gradient_cost)
+
+        # aggregated costs
+        with TensorflowGraph.shared_name_scope('aggregated', graph, name_scopes):
+          with tf.name_scope('gradient'):
+            loss = tf.add_n(gradient_costs)
+
+          # weight decay
+          if self.penalty != 0.0:
+            # using self-defined regularization
+            penalty = weight_decay(self.penalty_type, self.penalty)
+            loss += penalty
+
+      return loss 
+  
+  def cost(self, logits, labels, weights):
+    return tf.mul(tf.nn.sigmoid_cross_entropy_with_logits(logits, labels),
+                  weights)
+      
+  def add_output_ops(self, graph, output):
+    # adding output nodes of sigmoid function
+    with graph.as_default():
+      sigmoid = []
+      with tf.name_scope('inference'):
+        for i, logits in enumerate(output):
+          sigmoid.append(tf.nn.sigmoid(logits, name='sigmoid_%d' % i))
+      output = sigmoid
+    return output 
+    
+  def construct_feed_dict(self, X_b, y_b=None, w_b=None, ids_b=None):
+
+    orig_dict = {}
+    orig_dict["mol_features"] = X_b
+    for task in range(self.n_tasks):
+      if y_b is not None:
+        y_2column = to_one_hot(y_b[:, task])
+        # fix the size to be [?,1]
+        orig_dict["labels_%d" % task] = y_2column[:,1:2]
+      else:
+        # Dummy placeholders
+        orig_dict["labels_%d" % task] = np.zeros((self.batch_size,1))
+      if w_b is not None:
+        orig_dict["weights_%d" % task] = w_b[:, task]
+      else:
+        # Dummy placeholders
+        orig_dict["weights_%d" % task] = np.ones(
+            (self.batch_size,)) 
+    return TensorflowGraph.get_feed_dict(orig_dict)
+  
+  def predict_proba_on_batch(self, X, pad_batch=False):
```

Comment:
I believe `predict_proba_on_batch` will be defined via inheritance (should check this though)